File: leet/palindromeNumber.js
// Determine whether an integer is a palindrome. An integer is a palindrome when it reads the same backward as forward.

// Follow up: Could you solve it without converting the integer to a string?

/*
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
  if(x < 0){
      return false;
  }

  let number = x.toString()
  let start = 0;
  let end = number.length-1;
  while(start < end) {
      if(number[start] !== number[end]){
          return false;
      }
      start++;
      end --;
  }
  return true;
};

File: leet/palindromeNumber.py
# // Determine whether an integer is a palindrome. An integer is a palindrome when it reads the same backward as forward.

# // Follow up: Could you solve it without converting the integer to a string?

class Solution:
    def isPalindrome(self, x: int) -> bool:
        x = str(x)
        if(len(x) < 0):
            return False
        i = 0
        j = len(x)-1

        while i < j:
            if(x[i] != x[j]):
                return False
            i += 1
            j -= 1

        return True